requirements.txt

```diff
@@ -26,13 +26,13 @@ rules==3.0
 
 # These should be okay to update minors
 numpy==1.22.0
-pandas==1.3.1
+pandas==1.4.4
 pangres==2.3.1
```

Comment:
Seems like bumping `numpy` is also appropriate? `pangres` has had two major releases, so we may need to change API usages, but I'd look at that as well, since we're updating `pandas`.

Reply:
Looks like this `numpy` version is still compatible. And `pangres` is pretty far behind, but seems like it should theoretically still be compatible. Did you test the cron?

Can you rebase this with master so we can more effectively test with the latest version of cron (UDP)? Thanks

Reply:
When testing with UDW, I'm seeing this warning I wasn't seeing before when using `pd.read_sql`:
```
UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
```

Looks like we're passing a Django connection object. I guess it's probably better to create our own SQLAlchemy engine and then close it at the end of the cron?

See these:
- https://pandas.pydata.org/docs/reference/api/pandas.read_sql.html?highlight=read_sql#pandas.read_sql
- https://stackoverflow.com/a/71083448

Reply:
Actually, this is happening on the views, too. We'll need to think about how best to open/close connections. Maybe we can form a MySQL connection string? Alternatively, we can use the Django ORM to pull data in the views, and pass it into vanilla DataFrames, but that's a bigger change.

Reply:
I didn't test the cron.

Does the warning appear in the UI or in the server log?

If it doesn't appear in the UI, is the content affected by this warning?  Is it getting the wrong information?

If the UI is unaffected, no warning or content problem, what should we do?  Do we go ahead with this change on its own, or do we need to address the warning before including it in the release?

Reply:
I'd spend some time trying to address, and if it's a simple change, let's get it in. If not, we'll have to discuss more. Ideally, we don't have log warnings going off constantly as the tool runs, that affects our ability to inspect for other problems.

Reply:
To be clear, it's in the logs, the UI doesn't seem to be affected. It's a warning, not an error, but it seems like `pandas` is telling us, we're not explicitly supporting this type of connection object going forward.

Reply:
It sounds like some devs are just writing code to ignore it as Pandas mentions in the Issue that they want to officially support other raw SQL connections. I think we can open this as an issue and revisit.

https://github.com/pandas-dev/pandas/issues/45660#issuecomment-1077355514

Reply:
Okay well we only use this in the cron, views and data_validation.

We could put at the top before we import pandas.

```
import warnings
# ignore warning for non-SQLAlchemy Connection
# see github.com/pandas-dev/pandas/issues/45660
warnings.simplefilter('ignore', category=UserWarning)
```

We could create and mention a new issue there too. I don't think fixing this actually needs is a blocker for this release, but I'd like to get Python updated if it's the only thing holding it back. 

Rather than using simplefilter we could use filterwarnings to be more specific to just this message.

```
import warnings
# ignore warning for non-SQLAlchemy Connection
# see github.com/pandas-dev/pandas/issues/45660
warnings.filterwarnings('ignore', category=UserWarning, message='pandas only support SQLAlchemy connectable')
```

Reply:
I'm okay with a warning filter.

Reply:
Is this just a problem with the cron.py or were you seeing it in the application log from the views.py when using the `django.db.connection`? I guess I could run this and check.

I'm fine with either too, just seems like it's going to complicate this PR trying to implement something here for all the `read_sql` areas since we use 2 different databases.

Reply:
I'm seeing it in the views.

Reply:
Should be resolved by latest commits using the engine approach.

Reply:
I also bumped `numpy` and `pangres`, `pangres` only required a small change to respond to a parameter name change